doc/developer-guide/api/functions/TSRemap.en.rst

```diff
@@ -65,9 +66,14 @@ any data or continuations associated with that instance.
 entry point. In this function, the remap plugin may examine and modify
 the HTTP transaction.
 
-:func:`TSRemapConfigReload` is called once for every remap plugin immediately after a new
-configuration is successfully loaded and immediately before the new remap configuration becomes
-active. This is an optional entry point, which takes no arguments and has no return value.
+:func:`TSRemapPreConfigReload` is called *before* the parsing of a new remap configuration starts
```

Comment:
I'm not sure I'm in favor of this notification. There's a bit of vagueness about which plugins exactly are called, which doesn't happen in the post config reload case. For the latter, there is some set of plugins active in the remap, and the notification is called on each of those. For this, can the code look ahead to see which plugins will be loaded? This really should not be called on a plugin that's not going to be part of the new configuration.

Reply:
@SolidWallOfCode, I know that you already approved this but I kept thinking about your comment:

> This really should not be called on a plugin that's not going to be part of the new configuration.

I thought of this a little bit differently:

Most of the plugins is assumed to use per-plugin-instance data-structures
when reloading their configs and only a few of them that wish to optimize
performance or deal with the complexities of using per-plugin DSO "global"
data-structures would use plugin configuration reload notifications like
`TSRemapPreConfigReload` and `TSRemapPostConfigReload`.

Instead of trying to foresee the needs or the expectations of each use-case,
a more "open-ended" and straight-forward design was chosen for the configuration
reload notifications. The notifications are broadcasted to all loaded plugins
at the moments before and after the reload attempt, regardless of whether
a plugin is part of the new configuration or not.

During the `TSRemapPostConfigReload` we already decided to notify the
plugin if the plugin reload was successful or not so the plugins can
recover adequately in case of configuration reload failure.

Submitted PR #5790 that adds a signal to show if the particular plugin was part of
the new configuration. This will be beneficial for use-cases that expect
plugin reload notification not to be called in case they are not part of
the new configuration so they can ignore the notification.